src/test/java/org/springframework/data/elasticsearch/TestUtils.java

```diff
@@ -94,7 +94,7 @@ public static boolean isEmptyIndex(String indexName) {
 			return 0L == client
 					.search(new SearchRequest(indexName)
 							.source(SearchSourceBuilder.searchSource().query(QueryBuilders.matchAllQuery())), RequestOptions.DEFAULT)
-					.getHits().getTotalHits();
+					.getHits().getTotalHits().value;
```

Comment:
Let's add some utility class to fetch the total hits value. It's already bad design to let Lucene API leak into ES code. We should have a single utility method to fetch the total hits value instead spreading access to Lucene API across our code base.

Reply:
done that